File: services/iBoardService.go
package services

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"kanban-go/models"
)

type IBoardService interface {
	CreateBoard(BoardDto models.BoardDto) error
	GetBoards() ([]models.Board, error)
	GetBoardById(id primitive.ObjectID) (models.Board, error)
	UpdateBoard(Board models.Board) error
	DeleteBoard(id primitive.ObjectID) error
}

File: client/src/services/iBoardService.ts
import {boardType} from "@/models/boardType.ts";
import {ObjectId} from "bson";

export interface IBoardService {
    createBoard(boardName: string): Promise<boardType>;
    getBoards(): Promise<boardType[]>;
    getBoardById(id: ObjectId): Promise<boardType>;
    updateBoard(updatedBoard: boardType): string;
    deleteBoard(id: ObjectId): string;
}